pkgs/by-name/ht/htb-toolkit/package.nix

```diff
@@ -0,0 +1,85 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, openssl
+, stdenv
+, darwin
+, coreutils
+, noto-fonts-color-emoji
+, gnome
+, libsecret
+, bash
+, openvpn
+, nerdfonts
+, gzip
+, killall
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "htb-toolkit";
+  version = "unstable-2024-01-17";
+
+  src = fetchFromGitHub {
+    owner = "D3vil0p3r";
+    repo = "htb-toolkit";
+    # https://github.com/D3vil0p3r/htb-toolkit/issues/3
+    rev = "54e11774ea8746ea540548082d3b25c22306b4fc";
+    hash = "sha256-QYUqdqFV9Qn+VbJTnz5hx5I0XV1nrzCoCKtRS7jBLsE=";
+  };
+
+  cargoHash = "sha256-XDE6A6EIAUbuzt8Zb/ROfDAPp0ZyN0WQ4D1gWHjRVhg=";
+
+  # Patches only relevant for Nixpkgs
+  patches = [
+    ./main.patch
+    ./manage.patch
+    ./play.patch
+    ./types.patch
+    ./utils.patch
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    openssl
+  ] ++ lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
+
+  propagateBuildInputs = [
+    (nerdfonts.override { fonts = [ "NerdFontsSymbolsOnly" ]; })
+    coreutils
+    noto-fonts-color-emoji
+    gnome.gnome-keyring
+    libsecret
+    openvpn
+    gzip
+    killall
```

Comment:
Do all of these packages need to be propagated? I don't think whatever package depending on this package would need these binaries

Reply:
If `propagatedBuildInputs` allows to generate binaries of these dependencies, their binaries are all used except for gnome-keyring. But since I already used `substituteInPlace`, so I don't know if use `propagatedBuildInputs` and remove those `substituteInPlace` for getting binaries, or using `substituteInPlace` and removing `propagatedBuildInputs`.

Reply:
Considering most of these have already been wrapped into the binary, I don't think they would need to be propagated. Propagating build inputs like this is quite rare when working with compiled languages like rust.

It seems to me like they can be left out altogether, since they are injected into the source code in `postPatch`. Or if nothing else, normal `buildInputs`.

Reply:
Fixed

Reply:
Could you try testing it without putting it into `buildInputs` as well? They don't seem to be needed in `$PATH` since you patch the absolute paths into the source, and there's no reason to keep them in `buildInputs` unless they are needed as libraries. They are already implicitly dependencies of the derivation when you include them with `${...}` through a mechanism called [string context](https://shealevy.com/blog/2018/08/05/understanding-nixs-string-context/)

bintools `strings result/bin/htb-toolkit` when built without these listed in `buildInputs` seem to suggest the same thing.

Reply:
I tested it and it works correctly! Thank you for your suggestions @h7x4 !